packages/in_app_purchase/in_app_purchase_android/android/src/main/java/io/flutter/plugins/inapppurchase/Translator.java

```diff
@@ -26,7 +26,10 @@
     info.put("description", detail.getDescription());
     info.put("freeTrialPeriod", detail.getFreeTrialPeriod());
     info.put("introductoryPrice", detail.getIntroductoryPrice());
-    info.put("introductoryPriceAmountMicros", detail.getIntroductoryPriceAmountMicros());
+    // Make sure the `introductoryPriceAmountMicros` field is returned as a `String` value as the
+    // Dart side expects it as `String`. Changing the type in Dart would introduce a breaking change.
+    info.put(
+        "introductoryPriceAmountMicros", String.valueOf(detail.getIntroductoryPriceAmountMicros()));
```

Comment:
Changing this on the Java side will make a better fix later harder to do. Can't we change the transport name, and have a String getter on the dart side that returns the underlying int value as a String?

Reply:
@stuartmorgan I understand changing to int would technically be a breaking change, but only for people who read and use the field - and that field would *always* have returned an empty string. I therefore expect the number of builds that break to be very very small.

Is a compromise to change the field to an `int` and also add a getter `introductoryPriceAmountMicrosAsString` that does the conversion so the very few people who read the original field and expect a string can simply change their field to the new getter?

Reply:
> @stuartmorgan I understand changing to int would technically be a breaking change, but only for people who read and use the field 

I'm well aware that it's only a breaking change for people who reference the field; please re-read my comment.

> I therefore expect the number of builds that break to be very very small.

That may be true (or may not; you don't really know how many people did things like try to use it, find that it didn't seem to work for them, and added a fallback but left the code in place), but the definition of a non-breaking change in semantic versioning is not "I don't think it'll break many people".

> Is a compromise to change the field to an `int` and also add a getter `introductoryPriceAmountMicrosAsString` that does the conversion so the very few people who read the original field and expect a string can simply change their field to the new getter?

No, because that's a breaking change.

There is no way to change the type of the existing getter without it being a breaking change; this is not a matter of discussion, it's an objective fact. And we're not going to make an obviously breaking change without adjusting the version accordingly, because we do not deliberately violate semantic versioning in our plugins.

Reply:
Got it - didn't mean to come across as pushing, was just trying to help. I have absolutely no skin in this game (I came across the original error, I don't use the field at all).

If I may make one comment, it would be that the impact of the original change (that caused the error) was quite severe, in that it rendered the entire package unusable for Android users. Ideally, this would have been caught by tests before bumping the version. I am saying this because I had the same thing happen with the wakelock package just a few days ago, and if you have a flutter app with many package dependencies then these bugs as a result of version updates are very disruptive and difficult to discover and trace back to the offending package.  It leads me to consider locking packages to a specific version (instead of using the caret notation to benefit from upgrades), and that is not a good developer experience.

Reply:
> If I may make one comment, it would be that the impact of the original change (that caused the error) was quite severe, in that it rendered the entire package unusable for Android users. Ideally, this would have been caught by tests before bumping the version.

Yes, it's not clear to me yet why that PR passed our automated tests (of which we have quite a few). That's something we'll be looking into.

> I am saying this because I had the same thing happen with the wakelock package just a few days ago

That was a third-party package, and as far as I can tell from the nature of that bug, that plugin:
- has no automated testing of any kind, and
- was never launched on iOS after the change

We cannot control what third-party package authors do, but it was a very different scenario. That bug could not have passed our CI.